### PR TITLE
Provide Dockerfile files for both JVM and Native modes

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #1118

This is also sort of necessary if an AP4K guide (part of #1300) is to make sense and not confuse users

Supersedes #1173

### CODE REVIEW

**Comment by geoand** [comment]
@cescoffier are we going to be using this for the upcoming release?

File: devtools/common/src/main/resources/templates/dockerfile-jvm.ftl [hunk 1]
```diff
@@ -0,0 +1,19 @@
+####
+# Before building the docker image run:
```
Comment (cescoffier) [inline devtools/common/src/main/resources/templates/dockerfile-jvm.ftl]: Maybe add a line (before) explaining that this run the application in JVM mode. I know it's obvious, but well... you know ;-)

File: devtools/common/src/main/resources/templates/dockerfile-native.ftl [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #
 # Then, build the image with:
 #
-# docker build -f src/main/docker/Dockerfile -t quarkus/${project_artifactId} .
```
Comment (cescoffier) [inline devtools/common/src/main/resources/templates/dockerfile-native.ftl]: Same as the previous comment, we should add a line (first line) explaining that this runs the application in native mode.

**Review by cescoffier — CHANGES_REQUESTED** [review]
Almost there, I believe that 2 lines are required to better explain the difference in the Dockerfile.

File: devtools/common/src/main/resources/templates/dockerfile-jvm.ftl (hunk 1 above)
Comment (geoand) [reply devtools/common/src/main/resources/templates/dockerfile-jvm.ftl]: Sure thing!

**Comment by geoand** [comment]
@cescoffier I also need to remove the `0.0.0.0` stuff since that is now the default. On it now

File: devtools/common/src/main/resources/templates/dockerfile-jvm.ftl (hunk 1 above)
Comment (geoand) [reply devtools/common/src/main/resources/templates/dockerfile-jvm.ftl]: Done

File: devtools/common/src/main/resources/templates/dockerfile-native.ftl (hunk 2 above)
Comment (geoand) [reply devtools/common/src/main/resources/templates/dockerfile-native.ftl]: Done

**Comment by geoand** [comment]
@cescoffier PR updated and rebased onto the latest master (since it was far behind and I wanted to make sure that everything is still working)

**Comment by cescoffier** [comment]
Thanks, looks good now. Waiting for CI to merge it.

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
@geoand Just dicussed it with @stuartwdouglas, it might be better to keep the 0.0.0.0 for now, as it does not seems to be completely decided.

**Comment by geoand** [comment]
@cescoffier no problem, I will add it back in

**Comment by geoand** [comment]
@cescoffier PR updated to use `quarkus.http.host=0.0.0.0`

**Comment by geoand** [comment]
@cescoffier All green :tada:

**Comment by gsmet** [comment]
Merged, thanks!

**Comment by geoand** [comment]
You are welcome!

**Pull request merged**